pandas/plotting/_converter.py

```diff
@@ -247,6 +247,14 @@ def _convert_1d(values, units, axis):
         if isinstance(values, Index):
             return values.map(lambda x: get_datevalue(x, axis.freq))
         if is_period_arraylike(values):
+            # TODO: I don't understand what case is_period_arryalike
+            # is supposed to be handling. PeriodIndex has already
+            # been taken care of. I suspect it only handles Series[period]
```

Comment:
so what actually hits this now? IOW what if you remove this (and just leave the ``infer_dtype`` below)?

Reply:
I'll see if it's hit in the test suite. I'm nervous to remove it if it's not though, since I don't know how well tested this is.

Reply:
still I am puzzled why this would have much of a perf impact. and even your example only shows 50% improvement (when we need like 1000%) to bet back to the original, according to @jorisvandenbossche benchamrk.

Reply:
To be clear, the top of my post is against 0.23.4. So after this we're slightly faster than we were in 0.23.4

Reply:
oh I c, ok then.

But still I would take out the ``is_period_arraylike`` if we can

Reply:
Hmm, this goes back to https://github.com/pandas-dev/pandas/commit/7dd4091458d9117e57d2ad9ce3126855bd00108c (blame: https://github.com/pandas-dev/pandas/blame/1a266ee5809990244f1fe6daeb717878d06cf783/pandas/tseries/converter.py).

But actually, that was just a  slight change of https://github.com/pandas-dev/pandas/commit/c5af8e17937c21c8618498b3a2e1e886bd0813d6.

That points back to #9012, and that code snippet hits my new code, not the `is_period_arraylike(values)`. So I guess let's remove it.

Reply:
yeah the new path is much more general (and its performant actually)